src/tests/system/tests/test_ipa.py

```diff
@@ -390,3 +390,155 @@ def test_ipa__check_gssapi_authentication_indicator(client: Client, ipa: IPA):
+    """
+    # Create and apply the ID view.
+    ipa.idview("testview1").add(description="This is a new view")
+    ipa.idview("testview1").apply(hosts=[f"{client.host.hostname}"])
+
+    # Extract the attribute and its expected value from the override_attrs dictionary.
+    attr, expected_value = next(iter(override_attrs.items()))
+
+    # Apply the override using dictionary unpacking.
+    ipa.group("group-1").add().idgroupoverride().add_override("testview1", **override_attrs)
+    client.sssd.restart()
+
+    # If the attribute is "name", check both the original and the new group name.
+    groups_to_check = ["group-1", expected_value] if attr == "name" else ["group-1"]
+
+    # Verify that each group lookup returns the expected attribute value.
+    for group in groups_to_check:
+        result = client.tools.getent.group(group)
+        assert result is not None, f"group {group} not found in system lookup!"
+
+        # For attributes other than 'name', check that the attribute matches the expected value.
+        result_value = getattr(result, attr, None)
+        assert result_value == expected_value, f"Overridden {attr}: expected {expected_value}, got {result_value}"
+
+
+@pytest.mark.importance("high")
+@pytest.mark.topology(KnownTopology.IPA)
+def test_ipa__idview_append_user_cert(client: Client, ipa: IPA, moduledatadir: str):
+    """
+    :title: ID view overrides user certificate from file contents, but value is appended
+    :setup:
+        1. Create ID view and apply view to client
+        2. Add a user that overrides the user's certificate
+    :steps:
+        1. Look up user certificate
+    :expectedresults:
+        1. Certificate contains expected data and matches file contents
+    :customerscenario: False
+    """
+    ipa.idview("testview1").add(description="This is a new view")
+    ipa.idview("testview1").apply(hosts=[f"{client.host.hostname}"])
+
+    with open(f"{moduledatadir}/certificate") as f:
+        certificate_content = f.read().strip()
+
+    ipa.user("user-1").add().iduseroverride().add_override(
+        "testview1",
+        certificate=certificate_content,
+    )
+
+    client.sssd.restart()
+
+    result = ipa.user("user-1").iduseroverride().show_override("testview1")
+
+    assert certificate_content in result.get("usercertificate", [""])[0], "Certificate content mismatch!"
+
+
+@pytest.mark.importance("medium")
+@pytest.mark.topology(KnownTopology.IPA)
+def test_ipa__idview_fails_to_apply_on_ipa_master(ipa: IPA):
+    """
+    :title: ID views does not work on IPA master
+    :setup:
+        1. Add IPA ID view with description
+    :steps:
+        1. Apply ID view to IPA master
+    :expectedresults:
+        1. Applying ID view fails
+    :customerscenario: False
+    """
+    ipa.idview("testview1").add(description="This is a new view")
+    result = ipa.idview("testview1").apply(hosts=f"{ipa.host.hostname}")
+
+    assert result.rc == 1, "An IPA ID view should not apply on server"
```

Comment:
nitpick, ! at the end of the error message.

Reply:
added.

Reply:
I'm not seeing it. 

```    assert result.rc == 1, "An IPA ID view should not apply on server"```

Reply:
I did not push the changes, sorry, updated.